src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,123 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Entry<K, V>[] elementData;
+    private int size;
+
+    public MyHashMap() {
+        elementData = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > elementData.length * LOAD_FACTOR) {
+            resizeAndTransfer();
+        }
+
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+
+        int index = getIndexFor(key.hashCode());
```

Comment:
let's add checking to getIndexFor() method:
key == null - return 0 
and then we can remove method putForNullKey()

Reply:
Done